api/v1alpha1/snapshot_types.go

```diff
@@ -6,30 +6,40 @@
 package v1alpha1
 
 import (
-	"strings"
+	"fmt"
 
+	"github.com/mitchellh/hashstructure/v2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// Identity defines a cache entry. It is used to generate a hash that is then used by the
+// caching layer to identify an entry.
+type Identity map[string]string
```

Comment:
As this is part of the `spec` we need some validation on it. It is possible to flood `etcd` the way it is.

Reply:
That's a good point! Does `MaxItems` work on maps? 🤔 
```
// +kubebuilder:validation:MaxItems=10
```

Reply:
Nope. :(
`snapshot_types.go:18:15: must apply maxitem to an array`

Reply:
ah found it.
```
// +kubebuilder:validation:MaxProperties
```

Reply:
`+kubebuilder:validation:MaxProperties`